core/service.go

```diff
@@ -38,23 +38,28 @@ var _ services.Service = &Service{}
 // BABE session, and p2p service. It deals with the validation of transactions
 // and blocks by calling their respective validation functions in the runtime.
 type Service struct {
-	rt      *runtime.Runtime
-	bs      *babe.Session
-	blkRec  <-chan types.Block // receive blocks from BABE session
-	msgRec  <-chan p2p.Message // receive messages from p2p service
-	msgSend chan<- p2p.Message // send messages to p2p service
+	blockState   BlockState
+	storageState StorageState
+	rt           *runtime.Runtime
+	bs           *babe.Session
+	blkRec       <-chan types.Block // receive blocks from BABE session
```

Comment:
Do we still need this? If we have this, can we avoid passing state into the session?

Reply:
core needs the StorageState for other reasons, and I think it also needs blkRec still so that it can propagate blocks to p2p. otherwise, it would need to poll the state